permissions/src/test/java/org/odk/collect/permissions/PermissionsProviderTest.kt

```diff
@@ -23,118 +22,299 @@ class PermissionsProviderTest {
-            )
-        ).thenReturn(true)
+    fun `When camera permission is granted should requestCameraPermission call PermissionListener#granted`() {
+        permissionsApi.setGrantedPermission(Manifest.permission.CAMERA)
+
+        permissionsProvider.requestCameraPermission(activity, permissionListener)
+
+        verify(permissionListener).granted()
+        verifyNoMoreInteractions(permissionListener)
+        verifyNoInteractions(permissionsDialogCreator)
+    }
+
+    @Test
+    fun `When camera permission is not granted should requestCameraPermission call PermissionListener#denied`() {
+        permissionsProvider.requestCameraPermission(activity, permissionListener)
+
+        verify(permissionListener).denied()
+        verifyNoMoreInteractions(permissionListener)
+    }
+
+    @Test
+    fun `When camera permission is not granted should requestCameraPermission call PermissionsDialogCreator#showAdditionalExplanation`() {
+        permissionsProvider.requestCameraPermission(activity, permissionListener)
+
+        verify(permissionsDialogCreator).showAdditionalExplanation(
+            activity,
+            R.string.camera_runtime_permission_denied_title,
+            R.string.camera_runtime_permission_denied_desc,
+            R.drawable.ic_photo_camera,
+            permissionListener
+        )
+        verifyNoMoreInteractions(permissionsDialogCreator)
+    }
+
+    @Test
+    fun `When location permission granted should areLocationPermissionsGranted return true`() {
+        whenever(permissionsChecker.isPermissionGranted(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION)).thenReturn(true)
 
         assertThat(permissionsProvider.areLocationPermissionsGranted(), `is`(true))
     }
 
     @Test
-    fun `When location permissions not granted should areLocationPermissionsGranted() return false`() {
-        whenever(
-            permissionsChecker.isPermissionGranted(
-                Manifest.permission.ACCESS_FINE_LOCATION,
-                Manifest.permission.ACCESS_COARSE_LOCATION
-            )
-        ).thenReturn(false)
+    fun `When location permission is not granted should areLocationPermissionsGranted return false`() {
+        whenever(permissionsChecker.isPermissionGranted(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION)).thenReturn(false)
 
         assertThat(permissionsProvider.areLocationPermissionsGranted(), `is`(false))
     }
 
     @Test
-    fun `When camera and audio permissions granted should areCameraAndRecordAudioPermissionsGranted() return true`() {
-        whenever(
-            permissionsChecker.isPermissionGranted(
-                Manifest.permission.CAMERA,
-                Manifest.permission.RECORD_AUDIO
-            )
-        ).thenReturn(true)
+    fun `When location permission is granted and location is enabled in settings should requestEnabledLocationPermissions call PermissionListener#granted`() {
```

Comment:
Just in terms of test naming across these all of these, I think you can change `when <given> should <when> <then>` to just `when <given <when> <then>`. No need for "should". For instance, this test could be: `when location permission is granted and location is enabled in settings requestEnabledLocationPermissions calls PermissionListener#granted`. I sometimes find starting with the method under test can be clearer as well: `requestEnabledLocationPermissions() when location permission is granted and location is enabled in settings  calls PermissionListener#granted`.

Reply:
Fixed.